2765. Longest Alternating Subarray/solution.py

```diff
@@ -11,39 +10,81 @@ def alternatingSubarray(self, nums: List[int]) -> int:
 
     	# declare a subArraySize variable that keeps track of length of 
     	# alternating subarrays.
+        subArraySize = 0
 
-    	# declare a list to hold potential subArraySizes.
-
+    	# variable to hold the largest size
+        largestSize = 0
+		# variable to keep track of index in found subarray
+        subArrayIndex = 0
+        
+        # boolean to determine if found sub array in while loop iteration
+        foundSubArray = False
+        
     	# loop through the array starting from 0, increment index by 1 
     	# each time. this loop is the outer loop.
+        i = 0
+        while i < len(nums) - 1:
 
     		# if the int at next index is one greater than int at the 
     		# current index:
-
-    			# set subArraySize = 2 since we have an alternating subarray 
-    			# pair.
-
-    			# create a new variable subArrayIndex and set that equal to the 
-    			# current outer loop index + 1.
-
+            if nums[i] + 1 == nums[i+1]:
+                
+                # set the sub array size to 2 since we have 2 elements
+                subArraySize = 2
+                
+                # set subArrayIndex to keep track of where we are in sub array,
+                # add 2 since we're moving forward past the two elements we know
+                subArrayIndex = i + 2
+      
     			# check for continuation of the subarray via:
+                while subArrayIndex < len(nums):
+                    # if we're at an odd counter, check if current int is one 
+                    # less than previous
+                    if subArraySize % 2 != 0 and \
+                    nums[subArrayIndex] == nums[subArrayIndex - 1] + 1:
```

Comment:
By saving nums[subArrayIndex] - nums[subArrayIndex - 1] into a variable, you could have combined your two if clauses here and gotten rid of the code duplication where you increment the subArraySize